src/app/file/webp_format.cpp

```diff
@@ -0,0 +1,390 @@
+      free(buf);
+      return false;
+    }
+    if (fop_is_stop(fop))
+      break;
+  }
+
+  base::SharedPtr<WebPOptions> webPOptions = base::SharedPtr<WebPOptions>(new WebPOptions());
+  fop->seq.format_options = webPOptions;
+  webPOptions->lossless = std::min(config.input.format - 1, 1);
+
+  WebPIDelete(idec);
+  WebPFreeDecBuffer(&config.output);
+  return true;
+}
+
+#ifdef ENABLE_SAVE
+struct writerData {
+  FILE* fp;
+  FileOp* fop;
+};
+
+const std::pair<WebPEncodingError, std::string> enc_error_map_data[] = {
+    std::make_pair(VP8_ENC_OK, ""),
+    std::make_pair(VP8_ENC_ERROR_OUT_OF_MEMORY, "memory error allocating objects"),
+    std::make_pair(VP8_ENC_ERROR_BITSTREAM_OUT_OF_MEMORY, "memory error while flushing bits"),
+    std::make_pair(VP8_ENC_ERROR_NULL_PARAMETER, "a pointer parameter is NULL"),
+    std::make_pair(VP8_ENC_ERROR_INVALID_CONFIGURATION, "configuration is invalid"),
+    std::make_pair(VP8_ENC_ERROR_BAD_DIMENSION, "picture has invalid width/height"),
+    std::make_pair(VP8_ENC_ERROR_PARTITION0_OVERFLOW, "partition is bigger than 512k"),
+    std::make_pair(VP8_ENC_ERROR_PARTITION_OVERFLOW, "partition is bigger than 16M"),
+    std::make_pair(VP8_ENC_ERROR_BAD_WRITE, "error while flushing bytes"),
+    std::make_pair(VP8_ENC_ERROR_FILE_TOO_BIG, "file is bigger than 4G"),
+    std::make_pair(VP8_ENC_ERROR_OUT_OF_MEMORY, "memory error allocating objects"),
+    std::make_pair(VP8_ENC_ERROR_USER_ABORT, "abort request by user"),
+    std::make_pair(VP8_ENC_ERROR_LAST, "list terminator. always last.")
+};
+
+const std::map<WebPEncodingError, std::string> WebPEncodingErrorMap(enc_error_map_data, enc_error_map_data + sizeof enc_error_map_data / sizeof enc_error_map_data[0]);
+
+#if WEBP_ENCODER_ABI_VERSION < 0x0203
+#define MAX_LEVEL 9
+// Mapping between -z level and -m / -q parameter settings.
+static const struct {
+  uint8_t method_;
+  uint8_t quality_;
+} kLosslessPresets[MAX_LEVEL + 1] = {
+  { 0,  0 }, { 1, 20 }, { 2, 25 }, { 3, 30 }, { 3, 50 },
+  { 4, 50 }, { 4, 75 }, { 4, 90 }, { 5, 90 }, { 6, 100 }
+};
+int WebPConfigLosslessPreset(WebPConfig* config, int level) {
+  if (config == NULL || level < 0 || level > MAX_LEVEL) return 0;
+  config->lossless = 1;
+  config->method = kLosslessPresets[level].method_;
+  config->quality = kLosslessPresets[level].quality_;
+  return 1;
+}
+#endif
+
+static int ProgressReport(int percent, const WebPPicture* const pic)
+{
+  fop_progress(((writerData*)pic->custom_ptr)->fop, (double)percent/(double)100);
+  if (fop_is_stop(((writerData*)pic->custom_ptr)->fop)) return false;
+  return true;
+}
+
+static int FileWriter(const uint8_t* data, size_t data_size, const WebPPicture* const pic)
+{
+  return data_size ? (fwrite(data, data_size, 1, ((writerData*)pic->custom_ptr)->fp) == 1) : 1;
+}
+
+
+//Helper functions instead of std::stoi because of missing c++11 on mac os x
+template<typename T>
+static inline std::string ToString(const T& v)
```

Comment:
Sorry, it looks like a GitHub UI problem. I clicked the ToString() line. It looks like nobody is using it (anyway you can use `convert_to&lt;std::string&gt;()` if needed)